include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -4391,14 +4397,54 @@ namespace internal
 
     template <int fe_degree, int = 0>
     static bool
-    run(const unsigned int,
-        const AlignedVector<Number> &,
-        const Number *,
-        Number *,
+    run(const unsigned int                  n_desired_components,
+        const FEEvaluationBaseData<dim,
+                                   typename Number::value_type,
+                                   false,
+                                   Number> &fe_eval,
+        const Number *                      in_array,
+        Number *                            out_array,
         typename std::enable_if<fe_degree == -1>::type * = nullptr)
     {
       static_assert(fe_degree == -1, "Only usable for degree -1");
-      Assert(false, ExcNotImplemented());
+
+      const AlignedVector<Number> &inverse_shape =
+        fe_eval.get_shape_info().data.front().inverse_shape_values;
+
+      const unsigned int dofs_per_component =
+        fe_eval.get_shape_info().dofs_per_component_on_cell;
+      const unsigned int n_q_points = fe_eval.get_shape_info().n_q_points;
+
+      internal::
+        EvaluatorTensorProduct<internal::evaluate_general, dim, 0, 0, Number>
+          evaluator(AlignedVector<Number>(),
+                    AlignedVector<Number>(),
+                    inverse_shape,
```

Comment:
In the `evaluate_general` case, you do not necessarily need to put the `inverse_shape` to the `hessian` slot; in my opinion it would be slightly clearer to use the `values` function below. It is only the even-odd slot that does not have enough symmetries in the inverse shape values array to be compatible with the `values` case.

Reply:
Changed!